src/extensions/mvs/components/primitives.ts

```diff
@@ -582,10 +663,49 @@ function addLineMesh(context: PrimitiveBuilderContext, { groups, mesh }: MeshBui
         const count = Math.ceil(dist / (2 * params.dash_length));
         addFixedCountDashedCylinder(mesh, lStart, lEnd, 1.0, count, true, cylinderProps);
     } else {
+        // NOTE: this: use similar
         addSimpleCylinder(mesh, lStart, lEnd, cylinderProps);
     }
 }
 
+function _transformBoxLikePrimitive(center: PrimitivePositionT, extent: number[], scaling?: number[] | null, rotation_axis?: number[] | null, rotation_radians?: number | null, translation?: number[] | null) {
+    const mat4 = Mat4.identity();
+    const t = translation ?? [0, 0, 0];
+    if (isVector3(center)) {
+        // TODO: rotation
+        const translationVector = Vec3.create(center[0] + t[0], center[1] + t[1], center[2] + t[2]);
+        Mat4.translate(mat4, mat4, translationVector);
+    } else {
+        const translationVector = Vec3.create(boxPos[0] + t[0], boxPos[1] + t[1], boxPos[2] + t[2]);
+        Mat4.translate(mat4, mat4, translationVector);
+    }
+    // TODO: defaults if one of them is not provided?
+    if (rotation_axis && rotation_radians) {
+        const axis = Vec3.create(rotation_axis[0], rotation_axis[1], rotation_axis[2]);
+        Mat4.rotate(mat4, mat4, rotation_radians, axis);
+    }
+    const s = scaling ?? [1, 1, 1];
+    const scalingVector = Vec3.create(extent[0] * s[0], extent[1] * s[1], extent[2] * s[2]);
+    Mat4.scale(mat4, mat4, scalingVector);
+    return mat4;
+}
+
+const targetBox = Box3D.zero();
+
+function addBoxMesh(context: PrimitiveBuilderContext, { groups, mesh }: MeshBuilderState, node: MVSNode<'primitive'>, params: MVSPrimitiveParams<'box'>, options?: { skipResolvePosition?: boolean }) {
+    const box = Box();
+
+    if (!options?.skipResolvePosition) {
+        resolvePosition(context, params.center, boxPos, undefined, targetBox);
```

Comment:
Are there any places where this would be skipped?

Reply:
Not aware of any.